### PR TITLE
Adds a funny scene when getting Sisyphus achievement

### PR DESCRIPTION
**Pull request by intercepti0n**
## About The Pull Request

https://github.com/tgstation/tgstation/assets/137328283/979231c3-70cd-49dd-8d4d-2a8554adfbb3

## Why It's Good For The Game

This PR makes getting the “Ordeal of Sisyphus” extremely pointless by teleporting carried boulder and spessman back to the Lavaland. In my opinion, this feature is funny and suits the spirit of this achievement.

## Changelog

:cl:
add: Added a short scene when getting an Ordeal of Sisyphus achievement.
/:cl:

### CODE REVIEW

**Comment by Jacquerel** [comment]
This is very funny, but I believe it should be possible to engineer it such that the pod will take off _from_ your location, travel to Lavaland, and deposit the boulder there. This will make the camera transition less jarring and also allow other people to see what is going on.
I'll see if I can dig up the arguments required for podspawn if you can't figure it out.

**Comment by Sadboysuss** [comment]
love it

**Comment by intercepti0n** [comment]
@Jacquerel, I'll be glad 'cause I've got lost in supply pod code.

File: code/datums/components/sisyphus_awarder.dm [hunk 1]
```diff
@@ -45,5 +49,35 @@
...
+	start_reward_scene()
+
+/datum/component/sisyphus_awarder/proc/start_reward_scene()
+	var/list/turf/area_turfs = get_area_turfs(/area/lavaland/surface/outdoors)
+	var/list/target_turf
+
+	while (isnull(target_turf))
+		var/turf/possible_pick = pick(area_turfs)
+
+		if(possible_pick.density)
```
Comment (MrMelbert) [inline code/datums/components/sisyphus_awarder.dm:66]: You can also check for `isgroundlessturf` so we don't dump it into lava I guess

**Comment by Jacquerel** [comment]
> @Jacquerel, I'll be glad 'cause I've got lost in supply pod code.

Check `mechlaunchpad.dm` for an example.
You will want to set both the `"target"` and `"reverse_dropoff_coords"`.
You will likely need to make a subtype of the pod type you are using to set a couple of variables (`reversing`, `reverse_options_list`, `reverse_delays`)

I'd fuck around with the admin pod spawning menu until you have settings you like and then try and copy them over into the code.
Probably what you ideally want is for it to "land" instantly (so it seems to appear from your mob), take off quite quickly, and then spend a longer time landing the _second_ time (on Lavaland).

**Comment by Astrogem2** [comment]
spawning back at where you picked it up would also be pretty funny, but seems hard to code so

**Comment by Jacquerel** [comment]
> spawning back at where you picked it up would also be pretty funny, but seems hard to code so

It wouldn't actually be singificantly more difficult to put the person holding the boulder into the pod rather than the boulder

**Comment by newfren** [comment]
fuck now I need to get it again

**Comment by intercepti0n** [comment]
> spawning back at where you picked it up would also be pretty funny

That's a neat idea, @Astrogem2! I'll try implementing it.

**Comment by intercepti0n** [comment]
@Jacquerel , made a few corrections to the code and scene logic. It seems, I've finished.

**Comment by Jacquerel** [comment]
beautiful

**Pull request merged**